js/script.js

```diff
@@ -1,63 +1,185 @@
-
-var output = document.getElementById("welcome-message"); // komunikat powitalny
-
-if (!name || name === 'null'){
-  alert ("Nie podano imienia");
-} else {
-  output.innerHTML = "Witaj na stronie " + name + ". Poniżej znajdziesz przelicznik temperatury." + "<br>" + "Działa on w skali Fahrenheit'a i Celsiusza." + "<br>" + "Są to dwie używane obecnie skale tepmeratur."; // wypisanie komunikatu powitalnego po podaniu imienia
-}
-
-// Celsius to Fahrenheit
-function celsiusToFahrenheit (){
-  tempCelsius = window.prompt("Podaj temperaturę w stopniach Celsiusza"); // Prompt o podanie temperatury
-  if (tempCelsius.trim() !== null){ //Sprawdzenie czy podana wartość istnieje
-    if (parseFloat(tempCelsius) === "number" || !isNaN(tempCelsius)) { // Sprawdzenie czy wartość jest liczbą
-      tempFahrenheit = tempCelsius * 1.8 + 32;
-      outputFahrenheit.innerText = "Podana temperatura to " + tempCelsius + "℃. Jej odpowiednik to " + tempFahrenheit.toFixed(1) + "℉.";
-      if(tempCelsius < 0){
-        msg = "woda jest zamarznięta.";
-      } else if (tempCelsius >= 0 && tempCelsius < 100){
-        msg = "woda jest cieczą.";
-      } else if (tempCelsius >= 100){
-        msg = "woda zamienia się w parę.";
-      }
-      fahrenheitInfo.innerText = "Przy temperaturze " + tempCelsius + "℃, " + msg;
-    } else alert ("Musisz podać temperaturę!");
+// Global variables
+var celsiusToFahrenheitButton = document.getElementById("celsiusToFahrenheit"); // lewy przycisk
+var fahrenheitToCelsiusButton = document.getElementById("fahrenheitToCelsius"); // prawy przycisk
+var outputFirst = document.getElementById("outputFirst"); // wyświetlenie temperatury
+var outputSecond = document.getElementById("outputSecond"); // wyświetlenie informacji o stanie wody
+
+// First Button
+function getTemp() { // podanie temperatury
+  var temp = parseFloat(window.prompt("Podaj temperaturę:")); // zapisanie do zmiennej temp
+  return temp; // zwrócenie podanej wartości
+};
+
+
+function celsiusToFahrenheit(tempCelsius) { // funkcja do przeliczenia na st. Fahrenheita
+  var tempF = tempCelsius * 1.8 + 32; // przeliczenie na stopnie Far i zapisanie po przeliczniu do zmiennej tempF. To jest temperatura w Fahrenheitah
+  return tempF; // zwrócenie wartości temperatury w Fahrenheitah
+};
+
+function fahrenheitToCelsius(tempFahrenheit) { // funkcja do przeliczenia na st. Celsiusza
+  var tempC = (tempFahrenheit - 32) / 1.8; // przeliczenie na stopnie Cel i zapisanie po przeliczeniu do zmiennej tempC. Temperatura w Celsiuszach
+  return tempC; // zwrócenie wartości temperatury w Celsiuszach
+};
+
+function showTemp(temp){ // funkcja do wyświetlenia info z temperaturą
+  outputFirst.innerHTML = "Temperatura po przeliczeniu wynosi " + temp.toFixed(1) + " stopni";
+};
+
+fahrenheitToCelsiusButton.addEventListener("click", function() { // wywołanie po kolei funkcji dla pierwszego buttona
+  var tempF = getTemp();
+  checkTemp(tempF);
```

Comment:
konsola przegladarki mowi, ze `checkTemp is not defined` wiec teraz w miejsce `checkTemp` wstaw sprawdzanie, ktora miales wczesniej, czyli czy pobrana do zmiennej temperatura nie jest NaN i jesli nie jest, to wykonuj dalej akcje, ktore robisz, czyli showTemp i waterInfo, a jesli temperatura jest NaN, to wyswietl w outpucie informacje, ze temperatura nieprawidlowa